### PR TITLE
ability to set a default bearer token

### PR DESCRIPTION
**Pull request by michaelneale**
# Overview
Allow ability to have Bearer token optionally.

# Description
By setting an AUTH_TOKEN api calls can be checked against a bearer token.

# How Has This Been Tested?
A new test case for default behavior has been added.

## References
https://github.com/zalando/gin-oauth2

### CODE REVIEW

**Review by andresuribe87 — COMMENTED** [review]
I think we should do the introspection approach that was introduced in https://github.com/TBD54566975/ssi-service/pull/369/files (see the `introspect` middleware). We can separate that out into a single PR.

Integrating with an existing authorization server is how I expect production deployments to work.

**Review by decentralgabe — COMMENTED** [review]
I like this approach but would suggest that we pause on auth until we have a (light) [SIP](https://github.com/TBD54566975/ssi-service/tree/main/sip) in place that goes over options and examples.

**Comment by michaelneale** [comment]
@andresuribe87 yeah I like that - what we want to be careful of is shipping without something default on (as it will come back to bite) and default means assuming something pretty basic (but yeah a rel prod will integrate with many things).

**Comment by michaelneale** [comment]
cc @mistermoe @bradleydwyer if needed can easily add in

**Comment by decentralgabe** [comment]
I believe this should be merged, but since docs have shifted, @michaelneale can you move the README additions to a new piece of [service documentation here](https://github.com/TBD54566975/ssi-service/tree/main/doc#service-documentation) on auth?

**Comment by michaelneale** [comment]
@decentralgabe yep working on it...

File: pkg/server/server.go [hunk 1]
```diff
@@ -118,6 +118,7 @@ func setUpEngine(cfg config.ServerConfig, shutdown chan os.Signal) *gin.Engine {
 		gin.Recovery(),
 		gin.Logger(),
 		middleware.Errors(shutdown),
+		middleware.AuthMiddleware(),
```
Comment (decentralgabe) [inline pkg/server/server.go:121]: I think we want to comment this out

**Review by decentralgabe — APPROVED** [review]
nice

File: pkg/server/server.go (hunk 1 above)
Comment (michaelneale) [reply pkg/server/server.go:121]: it does skip it: https://github.com/TBD54566975/ssi-service/pull/556/files#diff-6320ecdc95f4d46f3bb6598130be227c00be32852a9774c073715363bae0c374R33 if not set

**Comment by michaelneale** [comment]
@decentralgabe ok to merge?

**Pull request merged**